sdk/nodejs/policy/deserialize.ts

```diff
@@ -0,0 +1,142 @@
+        // objects.
+        const sig: any = prop[specialSigKey];
+        if (sig) {
+            switch (sig) {
+                case specialAssetSig:
+                    if (prop["path"]) {
+                        return new asset.FileAsset(<string>prop["path"]);
+                    } else if (prop["text"]) {
+                        return new asset.StringAsset(<string>prop["text"]);
+                    } else if (prop["uri"]) {
+                        return new asset.RemoteAsset(<string>prop["uri"]);
+                    } else {
+                        throw new Error(
+                            "Invalid asset encountered when unmarshaling resource property",
+                        );
+                    }
+                case specialArchiveSig:
+                    if (prop["assets"]) {
+                        const assets: asset.AssetMap = {};
+                        for (const name of Object.keys(prop["assets"])) {
+                            const a = deserializeProperty(prop["assets"][name]);
+                            if (!asset.Asset.isInstance(a) && !asset.Archive.isInstance(a)) {
+                                throw new Error(
+                                    "Expected an AssetArchive's assets to be unmarshaled Asset or Archive objects",
+                                );
+                            }
+                            assets[name] = a;
+                        }
+                        return new asset.AssetArchive(assets);
+                    } else if (prop["path"]) {
+                        return new asset.FileArchive(<string>prop["path"]);
+                    } else if (prop["uri"]) {
+                        return new asset.RemoteArchive(<string>prop["uri"]);
+                    } else {
+                        throw new Error(
+                            "Invalid archive encountered when unmarshaling resource property",
+                        );
+                    }
+                case specialSecretSig:
+                    return deserializeProperty(prop["value"]);
+                default:
+                    throw new Error(
+                        `Unrecognized signature '${sig}' when unmarshaling resource property`,
+                    );
+            }
+        }
+
+        const obj: any = {};
+        for (const k of Object.keys(prop)) {
+            const o = deserializeProperty(prop[k]);
+            obj[k] = unwrapRpcSecret(o);
+        }
+
+        return obj;
+    }
+}
+
+/**
+ * isRpcSecret returns true if obj is a wrapped secret value (i.e. it's an object with the special
+ * key set).
+ */
+function isRpcSecret(obj: any): boolean {
+    return obj && obj[specialSigKey] === specialSecretSig;
+}
+
+/**
+ * unwrapRpcSecret returns the underlying value for a secret, or the value itself if it was not a
+ * secret.
+ */
+function unwrapRpcSecret(obj: any): any {
```

Comment:
This shouldn't be necessary, right? We're never returning a first-class secret from `deserializeProperty`, so all of them should already be unwrapped.

Reply:
We can get rid of both `isRpcSecret` and `unwrapRpcSecret`.